Common/Data/Custom/Intrinio/IntrinioEconomicDAta.cs

```diff
@@ -151,7 +153,9 @@ public override SubscriptionDataSource GetSource(SubscriptionDataConfig config,
                 subscription = new SubscriptionDataSource(url, SubscriptionTransportMedium.RemoteFile, FileFormat.Csv,
                                                           authorizationHeaders);
                 // Finally, here we force the engine to wait one second in case the we have multiples data sources added.
-                Thread.Sleep(millisecondsTimeout: 1000);
+                var intrinioApiCallLimit = 1000 - (DateTime.Now - LastIntrinoAPICall).Milliseconds;
+                if (intrinioApiCallLimit > 0) Thread.Sleep(millisecondsTimeout: intrinioApiCallLimit);
+                LastIntrinoAPICall = DateTime.Now;
```

Comment:
Maybe a better implementation is this:
```cs
var intrinioApiCallLimit = 1000 - (DateTime.Now - LastIntrinoAPICall).Milliseconds;
do
{
	if (intrinioApiCallLimit > 0) Thread.Sleep(intrinioApiCallLimit);
	intrinioApiCallLimit = 1000 - (DateTime.Now - LastIntrinoAPICall).Milliseconds;
} while (intrinioApiCallLimit > 0);
LastIntrinoAPICall = DateTime.Now;
```
To avoid race condition issues.

Reply:
@mchandschuh here what I mention below.